### ISSUE TITLE
[Feature] Disable warnings on Array.equals

### ISSUE THREAD

**Issue by jandk** [jqno/equalsverifier#1005]
### Describe the feature

When testing generated classes, we came across an issue where EqualsVerifier complained about arrays not using Arrays.equals. We could implement this, but we found no way to disable this warning.

We might just be missing something.

### Describe the solution you'd like

A warning we could disable specifically for Arrays

**Comment by jqno** [comment]
Hi!

I could add a way to disable the warning, and generated classes that you can't change could be a good reason to do so.

But I'm wondering, you say you could implement this. Does that mean you can change the code generation so that it does use Arrays.equals for the generated array? Because in general, that's what you want when comparing arrays. Using `equals` to compare arrays is like using `==` to compare strings; it's only true when they are the same object instance.

**Comment by jandk** [comment]
Oops, completely forgot I made this ticket...

We were abusing the `ObjectMethods.bootstrap` method to generate record-style equals/hashCode and toString. They don't call `Array.equals` but just use the identity (which is already slightly questionable, but I guess records are for immutable types, and arrays are mutable so you're not supposed to put them in a record in the first place).

Since then we changed and added an escape, if we have arrays and the bootstrap method fails, we do check if the array contents are actually equal. This makes equalsVerifier happy, but it still might make a nice addition nonetheless.

It's not a showstopper anymore, but it might be nice. Up to your discretion.

Greetings
